File: aoj/itp1/topic4/4_d_min_max_and_sum.java
import java.util.Scanner;

class Main {
  public static void main(String[] args) {
    int min = 1000000;
    int max = -1000000;
    long sum = 0;
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    for (int i = 0; i < n; i++) {
      int a = sc.nextInt();
      min = Math.min(min, a);
      max = Math.max(max, a);
      sum += a;
    }
    System.out.println(min + " " + max + " " + sum);
  }
}

File: aoj/itp1/topic4/4_d_min_max_and_sum.cc
#include <iostream>

int main() {
  int n;
  std::cin>> n;
  int min = 1000000;
  int max = -1000000;
  int64_t sum = 0;
  int a;
  for (int i = 0; i < n; ++i) {
    std::cin >> a;
    min = std::min(min, a);
    max = std::max(max, a);
    sum += a;
  }
  std::cout << min << " " << max << " " << sum << std::endl;
}
